compiler/circle-partitioner/README.md

```diff
@@ -0,0 +1,115 @@
+# circle-partitioner
+
+_circle-partitioner_ provides model partitioning of circle model to two or more circle models.
+
+### Available backends in OneRT
+
+- `cpu`, `acl_cl`, `acl_neon`, `ruy`, `xnnpack`
+
+### Execution Interface (preliminary)
+
+Let's try with this interface:
+```
+circle_partitioner --partition=Net_InstanceNorm_003.part \
+   --backends=cpu,acl_cl --default=cpu \
+   Net_InstanceNorm_003.circle \
+   Net_InstanceNorm_003.part.circle
+```
+
+where `Net_InstanceNorm_003.part` is like this for initial design
+```
+[partition]
+backends=cpu,acl_cl
+default=cpu
+comply=opcode
+
+[OPCODE]
+ADD=acl_cl
+```
+where in `[partition]` section,
+- `backends` is available backends and can be overridden by `--backends`
+- `default` is default backend for OpCodes not assigned in `[OPCODE]` section can be overridden by `--default`
+- `comply` is which rule to apply, where only `opcode` is available for now
+
+Alternative thoughts (1)
+- `_=` is to assign default
+```
+[OPCODE]
+_=cpu
+ADD=acl_cl
+```
+
+Alternative thoughts (2)
+- Use Op name to assign backend
+```
+[OP]
+Reduction_indices=GPU
+```
+- there are very long names that may be inconvenient...
+
+### Partitioned output (preliminary)
+
+#### Output files
+
+After partition is applied, output files would look something like these
+- `Net_InstanceNorm_003.part.00001_cpu.circle`
+- `Net_InstanceNorm_003.part.00002_acl_cl.circle`
+- `Net_InstanceNorm_003.part.00003_cpu.circle`
+- `Net_InstanceNorm_003.part.config`
+
+Assume only `Div` node is assigned to `acl_cl`
+
+#### Connection (preliminary)
+
+Format candidate 1:
+- `Net_InstanceNorm_003.part.config` provides connection of each circle files.
+```
+[source]
+file=Net_InstanceNorm_003.circle
+i1=Input
+o1=Add_as_terminal
+
+[models]
+m1=Net_InstanceNorm_003.part.00001_cpu.circle
+m2=Net_InstanceNorm_003.part.00002_acl_cl.circle
+m3=Net_InstanceNorm_003.part.00003_cpu.circle
+
+[Net_InstanceNorm_003.part.00001_cpu.circle]
+i1=Input
+o1=Pow
+o2=Sub
+
+[Net_InstanceNorm_003.part.00002_acl_cl.circle]
+i1=Sub@Net_InstanceNorm_003.part.00002_acl_cl.circle
+i2=Pow@Net_InstanceNorm_003.part.00002_acl_cl.circle
+o1=Div
+
+[Net_InstanceNorm_003.part.00003_cpu.circle]
+i1=Div@Net_InstanceNorm_003.part.00002_acl_cl.circle
+o1=Add_as_terminal
+```
+
+Predefined section
+- `source`: Source circle model information. Has `file` as filename, `iN` for inputs and `oN` for outputs.
+- `models`: Partitioned circle models. Has `mN` for model filename.
+- `source` can be optional?
```

Comment:
At least, `iN` and `oN` in `source` section is required for runtime to map input/output buffer from user to partitioned model.